sql/server.go

```diff
@@ -186,13 +211,32 @@ func (s *Server) exec(req driver.Request) (resp driver.Response, err error) {
 			return
 		}
 		var plan planNode
-		if err = s.db.Txn(func(txn *client.Txn) error {
-			planMaker.txn = txn
+		if _, ok := stmt.(*parser.BeginTransaction); ok {
+			if planMaker.txn != nil {
+				err = errBeginDuringOpenTxn
+				return
+			}
+			// Start a transaction here and not in planMaker because planMaker
+			// doesn't have access to the db object.
```

Comment:
We could give planMaker access to the db object. I think the better reason to do this here is to avoid the auto-transaction creation that occurs when a transaction doesn't already exist.